flang/lib/Lower/Bridge.cpp

```diff
@@ -1714,7 +1715,8 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     genEndChangeTeamStmt(*this, getEval(), stmt);
   }
 
-  void genFIR(const Fortran::parser::CriticalConstruct &) {
+  void genFIR(const Fortran::parser::CriticalConstruct &criticalConstruct) {
+    setCurrentPosition(Fortran::parser::FindSourceLocation(criticalConstruct));
```

Comment:
This pattern is pretty repetitive. Maybe use a template function?

Reply:
Good idea, done.